### ISSUE TITLE
provider/aws: destroy fails on instances with aws_volume_attachment defined

### ISSUE THREAD

**Issue by discogestalt** [hashicorp/terraform#4643]
tldr:
running `terraform destroy` against an AWS instance should not try to detach an EBS volume before terminating the instance.

I'm building a test infrastructure and then tearing it back down, but the destroy command fails when I have instances with attached EBS volumes. Terminating an instance with attached storage is perfectly valid, but terraform doesn't allow it because it tries to detach the volume before terminating the instance, but that operation fails because the volume is mounted on a running instance... It shouldn't even try to detach the storage when and instance is being terminated, there's really no point to that.

My instances are defined as:

```
resource "aws_instance" "my_instance" {
  ami = "${var.ami}"
  instance_type = "m4.xlarge"
  key_name = "${var.key_name}"
  vpc_security_group_ids = ["${var.security_group}"]
  subnet_id = "${var.subnet}"
  root_block_device = {
    delete_on_termination = true
  }
}
```

I then define an EBS volume and attach it:

```
resource "aws_ebs_volume" "my_volume" {
  availability_zone = "${aws_instance.my_instance.availability_zone}"
  type = "gp2"
  size = 100
  tags = {
    Name = "hostname/mount"
    mount_point = "/mount"
  }
}

resource "aws_volume_attachment" "my_volume" {
  device_name = "/dev/sdf"
  volume_id = "${aws_ebs_volume.my_volume.id}"
  instance_id = "${aws_instance.my_instance.id}"
  provisioner "remote-exec" {
    script = "${path.root}/../scripts/disk_finish.sh"
  }
}
```

The `disk_finish.sh` script in the provisioner updates /etc/fstab, runs mkfs on the volume and then mounts it.

**Comment by arthurschreiber** [comment]
See #2957 for a similar problem, and the feature requrest in #622 which would provide an easy fix for this.

**Comment by phinze** [comment]
Thanks @discogestalt for the report and to @arthurschreiber for linking these issues together. I'm going to close this one in favor of #2957 to consolidate the conversation and I'll follow up on that thread.
